langkit/dsl_unparse.py

```diff
@@ -1350,7 +1356,7 @@ def emit_node_type(node_type):
         builtin_properties = node_type.builtin_properties()
 
         if node_type.is_root_node:
-            base_name = "Node"
+            traits.append(f'Node[{type_name(node_type)}]')
```

Comment:
I can’t find an issue in the updated baselines, so I suppose this part is not exercized yet, but this looks wrong to me: shouldn’t we just use `type_name(node_type)` to designate the root node type? `Node[…]` looks just like an  instantiated trait (the one that the root node implements), which is different.

Reply:
Not sure to get your point here.

What we want now for the root node is to print it as a trait implementing `Node`, e.g.: `class FooNode implements Node[FooNode]` in place of `class FooNode : Node` for instance.

Reply:
Ah yes, my bad! I thought the purpose of this part of the code was to emit a type reference to `node_type` instead of emitting its type declaration. So this is actually fine, sorry!